DependencyInjection/Compiler/AddProcessorsPass.php

```diff
@@ -46,13 +48,15 @@ public function process(ContainerBuilder $container)
                     $definition = $container->getDefinition('monolog.logger_prototype');
                 }
 
-                if (!empty($tag['method'])) {
-                    $processor = [new Reference($id), $tag['method']];
-                } else {
-                    // If no method is defined, fallback to use __invoke
-                    $processor = new Reference($id);
+                if (1 === Logger::API || is_a($definition->getClass(), ProcessableHandlerInterface::class, true)) {
```

Comment:
in case `empty($tag['handler'])`, `$definition` might contain an instance of `Logger` which is neither Version1 nor `ProcessableHandlerInterface`.